File: language-support/js/daml-ledger/src/model/get_transaction_by_id_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

/**
 * @interface GetTransactionByIdRequest
 * @memberof ledger
 */
export interface GetTransactionByIdRequest {
    /**
     * @member {string} transactionId
     * @memberof ledger.GetTransactionByIdRequest
     * @instance
     */
    transactionId: string
    /**
     * @member {Array<string>} requestingParties
     * @memberof ledger.GetTransactionByIdRequest
     * @instance
     */
    requestingParties: string[]
}

File: canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/messages/transaction/GetTransactionByIdRequest.scala
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.ledger.api.messages.transaction

import com.daml.lf.data.Ref.Party
import com.digitalasset.canton.ledger.api.domain.{LedgerId, TransactionId}

final case class GetTransactionByIdRequest(
    ledgerId: Option[LedgerId],
    transactionId: TransactionId,
    requestingParties: Set[Party],
)
